tests/ui/test_ui_tools.py

```diff
@@ -2221,11 +2224,11 @@ def test_msg_generates_search_and_header_bar(
         "expected_header, to_vary_in_last_message",
         [
             (
-                [STATUS_INACTIVE, "alice", " ", "DAYDATETIME"],
-                {"sender_full_name": "bob"},
+                [STATUS_INACTIVE, "alice (4)", " ", "DAYDATETIME"],
+                {"sender_id": "4"},
             ),
             ([" ", " ", " ", "DAYDATETIME"], {"timestamp": 1532103779}),
-            ([STATUS_INACTIVE, "alice", " ", "DAYDATETIME"], {"timestamp": 0}),
+            ([STATUS_INACTIVE, "alice (4)", " ", "DAYDATETIME"], {"timestamp": 0}),
```

Comment:
So I tried changing it back to normal (i.e. name without user id) but it was still passing the duplicate name condition and appending `None`. At first I thought the test user_list has `alice` as a common name but changing it to a unique name also gave the same results. Then I realised the `this_msg` and `last_msg` in the test function are derived from the same parameters and they have the same sender name (although different `sender_id`) and that's why its always appending the `sender_id` to the name.

Reply:
I looked into this a bit more and realised there's no way to include the duplicate username case in the earlier function where we test headers  without appending the userid to every name in the parameters because we are keeping the same message and varying just a few attributes. Also, added a mock response as `false` for the `is_user_name_duplicate` function so that it doesn't fail for the reasons mentioned earlier, and since duplicate usernames aren't what we are testing in that test anyway.

Finally, I decided to make a new test for testing the specific case where username is a duplicate where it would need userid appended to it.